# Analyzing NYC SAT scores

In [16]:
import pandas as pd
import numpy as np

In [17]:
# Reading in all the data in dictionary
data_files = [
    "ap_2010.csv",
    "class_size.csv",
    "demographics.csv",
    "graduation.csv",
    "hs_directory.csv",
    "sat_results.csv"
]

data = {}
    
for f in data_files:
    d = pd.read_csv("schools/{0}".format(f))
    key_name = f.replace(".csv", "")
    data[key_name] = d

** Let's explore SAT results as it contains information for every high school in New York City **

In [18]:
data["sat_results"].head(5)

DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  Num of SAT Test Takers SAT Critical Reading Avg. Score SAT Math Avg. Score  \
0                     29                             355                 404   
1                     91                             383                 423   
2                     70                             377                 402   
3                      7                             414                 401   
4                     44                             390                 433   

  SAT Writing Avg. Score  
0                    363  
1                    366  
2                    370  
3                    359  
4                    384

From SAT results data, it seems like `DBN` is unique for each high school. We have scores for for Reading, Writing and Math.

In [19]:
for k,v in data.items():
    print("\n" + k + "\n")
    print(v.head(3))


class_size

   CSD BOROUGH SCHOOL CODE                SCHOOL NAME GRADE  PROGRAM TYPE  \
0    1       M        M015  P.S. 015 Roberto Clemente     0K       GEN ED   
1    1       M        M015  P.S. 015 Roberto Clemente     0K          CTT   
2    1       M        M015  P.S. 015 Roberto Clemente     01       GEN ED   

  CORE SUBJECT (MS CORE and 9-12 ONLY) CORE COURSE (MS CORE and 9-12 ONLY)  \
0                                    -                                   -   
1                                    -                                   -   
2                                    -                                   -   

  SERVICE CATEGORY(K-9* ONLY)  NUMBER OF STUDENTS / SEATS FILLED  \
0                           -                               19.0   
1                           -                               21.0   
2                           -                               17.0   

   NUMBER OF SECTIONS  AVERAGE CLASS SIZE  SIZE OF SMALLEST CLASS  \
0                 1.0  

`DBN` is available as column with unique values in every dataframe. This is how, we'll combine all of our datasets into one keeping `DBN` as primary key value. 

** Reading survey data **

Survey data is available in tab delimited formate. Hence, we'll be using encoding value to read it. Once done, let's combine to single dataframe to make it easier to read.

In [20]:
all_survey = pd.read_csv("schools/survey_all.txt", delimiter="\t", encoding="windows-1252")
d75_survey = pd.read_csv("schools/survey_d75.txt", delimiter="\t", encoding="windows-1252")

survey = pd.concat([all_survey, d75_survey], axis=0)
survey.head(3)

N_p  N_s   N_t  aca_p_11  aca_s_11  aca_t_11  aca_tot_11    bn  com_p_11  \
0   90.0  NaN  22.0       7.8       NaN       7.9         7.9  M015       7.6   
1  161.0  NaN  34.0       7.8       NaN       9.1         8.4  M019       7.6   
2  367.0  NaN  42.0       8.6       NaN       7.5         8.0  M020       8.3   

   com_s_11   ...    t_q8c_1  t_q8c_2  t_q8c_3 t_q8c_4  t_q9  t_q9_1  t_q9_2  \
0       NaN   ...       29.0     67.0      5.0     0.0   NaN     5.0    14.0   
1       NaN   ...       74.0     21.0      6.0     0.0   NaN     3.0     6.0   
2       NaN   ...       33.0     35.0     20.0    13.0   NaN     3.0     5.0   

   t_q9_3  t_q9_4  t_q9_5  
0    52.0    24.0     5.0  
1     3.0    78.0     9.0  
2    16.0    70.0     5.0  

[3 rows x 2773 columns]

Since there are 2773 columns, let's get only necessary columns based off [data dictionary from download location](https://data.cityofnewyork.us/Education/NYC-School-Survey-2011/mnz3-dyi8)

In [21]:
# Changing case of dbn to make it easier to merge
survey["DBN"] = survey["dbn"]

# Required columns
columns = ["DBN", "rr_s", "rr_t", "rr_p", "N_s", "N_t", "N_p", "saf_p_11", "com_p_11", "eng_p_11", "aca_p_11", "saf_t_11", "com_t_11", "eng_t_10", "aca_t_11", "saf_s_11", "com_s_11", "eng_s_11", "aca_s_11", "saf_tot_11", "com_tot_11", "eng_tot_11", "aca_tot_11",]

survey= survey.loc[:,columns]
data["survey"] = survey

survey.head(3)

DBN  rr_s  rr_t  rr_p  N_s   N_t    N_p  saf_p_11  com_p_11  eng_p_11  \
0  01M015   NaN    88    60  NaN  22.0   90.0       8.5       7.6       7.5   
1  01M019   NaN   100    60  NaN  34.0  161.0       8.4       7.6       7.6   
2  01M020   NaN    88    73  NaN  42.0  367.0       8.9       8.3       8.3   

      ...      eng_t_10  aca_t_11  saf_s_11  com_s_11  eng_s_11  aca_s_11  \
0     ...           NaN       7.9       NaN       NaN       NaN       NaN   
1     ...           NaN       9.1       NaN       NaN       NaN       NaN   
2     ...           NaN       7.5       NaN       NaN       NaN       NaN   

   saf_tot_11  com_tot_11  eng_tot_11  aca_tot_11  
0         8.0         7.7         7.5         7.9  
1         8.5         8.1         8.2         8.4  
2         8.2         7.3         7.5         8.0  

[3 rows x 23 columns]

** Cleaning up other data files **

Some of the other data files need to cleaned up as they don't have `DBN` column.

In [22]:
data["class_size"].head(3)

CSD BOROUGH SCHOOL CODE                SCHOOL NAME GRADE  PROGRAM TYPE  \
0    1       M        M015  P.S. 015 Roberto Clemente     0K       GEN ED   
1    1       M        M015  P.S. 015 Roberto Clemente     0K          CTT   
2    1       M        M015  P.S. 015 Roberto Clemente     01       GEN ED   

  CORE SUBJECT (MS CORE and 9-12 ONLY) CORE COURSE (MS CORE and 9-12 ONLY)  \
0                                    -                                   -   
1                                    -                                   -   
2                                    -                                   -   

  SERVICE CATEGORY(K-9* ONLY)  NUMBER OF STUDENTS / SEATS FILLED  \
0                           -                               19.0   
1                           -                               21.0   
2                           -                               17.0   

   NUMBER OF SECTIONS  AVERAGE CLASS SIZE  SIZE OF SMALLEST CLASS  \
0                 1.0                19.0                    19.0   
1                 1.0                21.0                    21.0   
2                 1.0                17.0                    17.0   

   SIZE OF LARGEST CLASS DATA SOURCE  SCHOOLWIDE PUPIL-TEACHER RATIO  
0                   19.0         ATS                             NaN  
1                   21.0         ATS                             NaN  
2                   17.0         ATS                             NaN

It can be seen that `DBN` column can be created by combining `CSD` and `SCHOOL CODE` columns.

In [23]:
data["hs_directory"]["DBN"] = data["hs_directory"]["dbn"]

def padded_csd(num):
    string_num = str(num)
    if len(string_num) > 1:
        return string_num
    else:
        return "0" + string_num
    
data["class_size"]["padded_csd"] = data["class_size"]["CSD"].apply(padded_csd)
data["class_size"]["DBN"] = data["class_size"]["padded_csd"] + data["class_size"]["SCHOOL CODE"]

data["class_size"].head(3)

CSD BOROUGH SCHOOL CODE                SCHOOL NAME GRADE  PROGRAM TYPE  \
0    1       M        M015  P.S. 015 Roberto Clemente     0K       GEN ED   
1    1       M        M015  P.S. 015 Roberto Clemente     0K          CTT   
2    1       M        M015  P.S. 015 Roberto Clemente     01       GEN ED   

  CORE SUBJECT (MS CORE and 9-12 ONLY) CORE COURSE (MS CORE and 9-12 ONLY)  \
0                                    -                                   -   
1                                    -                                   -   
2                                    -                                   -   

  SERVICE CATEGORY(K-9* ONLY)  NUMBER OF STUDENTS / SEATS FILLED  \
0                           -                               19.0   
1                           -                               21.0   
2                           -                               17.0   

   NUMBER OF SECTIONS  AVERAGE CLASS SIZE  SIZE OF SMALLEST CLASS  \
0                 1.0                19.0                    19.0   
1                 1.0                21.0                    21.0   
2                 1.0                17.0                    17.0   

   SIZE OF LARGEST CLASS DATA SOURCE  SCHOOLWIDE PUPIL-TEACHER RATIO  \
0                   19.0         ATS                             NaN   
1                   21.0         ATS                             NaN   
2                   17.0         ATS                             NaN   

  padded_csd     DBN  
0         01  01M015  
1         01  01M015  
2         01  01M015

** Combining all the 3 SAT scores to one **

In [24]:
# Using pandas.to_numeric to convert values to numeric 
courses = ["SAT Math Avg. Score", "SAT Critical Reading Avg. Score", "SAT Writing Avg. Score"]

for c in courses:
    data["sat_results"][c] = pd.to_numeric(data["sat_results"][c], errors="coerce")

data["sat_results"]["sat_score"] = data["sat_results"][courses[0]] + data["sat_results"][courses[1]] + data["sat_results"][courses[2]]

data["sat_results"]["sat_score"].head(3)

0    1122.0
1    1172.0
2    1149.0
Name: sat_score, dtype: float64

** Getting coordinates for each school for maps **

In [25]:
import re

def get_latitude(address):
    lat_long =  re.findall("\(.+, .+\)", address)
    return lat_long[0].split(", ")[0].replace("(", "")
    
data["hs_directory"]["lat"] = data["hs_directory"]["Location 1"].apply(get_latitude)

def get_longitude(address):
    lat_long =  re.findall("\(.+, .+\)", address)
    return lat_long[0].split(",")[1].replace(")", "").strip()
    
data["hs_directory"]["lon"] = data["hs_directory"]["Location 1"].apply(get_longitude)

data["hs_directory"]["lat"] = pd.to_numeric(data["hs_directory"]["lat"], errors="coerce")
data["hs_directory"]["lon"] = pd.to_numeric(data["hs_directory"]["lon"], errors="coerce")

data["hs_directory"].head(3)

dbn                                  school_name      boro  \
0  17K548          Brooklyn School for Music & Theatre  Brooklyn   
1  09X543             High School for Violin and Dance     Bronx   
2  09X327  Comprehensive Model School Project M.S. 327     Bronx   

  building_code  phone_number    fax_number grade_span_min  grade_span_max  \
0          K440  718-230-6250  718-230-6262              9              12   
1          X400  718-842-0687  718-589-9849              9              12   
2          X240  718-294-8111  718-294-8109              6              12   

  expgrade_span_min  expgrade_span_max    ...      \
0               NaN                NaN    ...       
1               NaN                NaN    ...       
2               NaN                NaN    ...       

                        priority05 priority06 priority07 priority08  \
0                              NaN        NaN        NaN        NaN   
1                              NaN        NaN        NaN        NaN   
2  Then to New York City residents        NaN        NaN        NaN   

  priority09  priority10                                         Location 1  \
0        NaN         NaN  883 Classon Avenue\nBrooklyn, NY 11225\n(40.67...   
1        NaN         NaN  1110 Boston Road\nBronx, NY 10456\n(40.8276026...   
2        NaN         NaN  1501 Jerome Avenue\nBronx, NY 10452\n(40.84241...   

      DBN        lat        lon  
0  17K548  40.670299 -73.961648  
1  09X543  40.827603 -73.904475  
2  09X327  40.842414 -73.916162  

[3 rows x 61 columns]

## Combining all the datasets

Since data clean up is done, we can start combining all the datasets into one by keeping `DBN` as unique column.

** Condensing class_size **

`class_size` dataset has multiple rows for same school based on `GRADE`, `PROGRAM TYPE` etc. We can get rid of lot of rows as we are only interested in in grades 9 through 12 and where type is Gen Ed.

In [26]:
class_size = data["class_size"]
class_size = class_size[class_size["GRADE "] == "09-12"]
class_size = class_size[class_size["PROGRAM TYPE"] == "GEN ED"]

class_size.head(3)

CSD BOROUGH SCHOOL CODE                                    SCHOOL NAME  \
225    1       M        M292  Henry Street School for International Studies   
226    1       M        M292  Henry Street School for International Studies   
227    1       M        M292  Henry Street School for International Studies   

    GRADE  PROGRAM TYPE CORE SUBJECT (MS CORE and 9-12 ONLY)  \
225  09-12       GEN ED                              ENGLISH   
226  09-12       GEN ED                              ENGLISH   
227  09-12       GEN ED                              ENGLISH   

    CORE COURSE (MS CORE and 9-12 ONLY) SERVICE CATEGORY(K-9* ONLY)  \
225                           English 9                           -   
226                          English 10                           -   
227                          English 11                           -   

     NUMBER OF STUDENTS / SEATS FILLED  NUMBER OF SECTIONS  \
225                               63.0                 3.0   
226                               79.0                 3.0   
227                               38.0                 2.0   

     AVERAGE CLASS SIZE  SIZE OF SMALLEST CLASS  SIZE OF LARGEST CLASS  \
225                21.0                    19.0                   25.0   
226                26.3                    24.0                   31.0   
227                19.0                    16.0                   22.0   

    DATA SOURCE  SCHOOLWIDE PUPIL-TEACHER RATIO padded_csd     DBN  
225       STARS                             NaN         01  01M292  
226       STARS                             NaN         01  01M292  
227       STARS                             NaN         01  01M292

`DBN` value is still not unique due to `CORE SUBJECT` columns. These two columns hold values for different core subjects offered. Let's take average across all the classes a school offers. This will give us unique `DBN` values, but also incorporate as much data as possible into the average

In [27]:
class_size = class_size.groupby("DBN").agg(np.mean)
class_size.reset_index(inplace=True)
data["class_size"] = class_size

data["class_size"].head(3)

DBN  CSD  NUMBER OF STUDENTS / SEATS FILLED  NUMBER OF SECTIONS  \
0  01M292    1                               88.0                 4.0   
1  01M332    1                               46.0                 2.0   
2  01M378    1                               33.0                 1.0   

   AVERAGE CLASS SIZE  SIZE OF SMALLEST CLASS  SIZE OF LARGEST CLASS  \
0           22.564286                    18.5              26.571429   
1           22.000000                    21.0              23.500000   
2           33.000000                    33.0              33.000000   

   SCHOOLWIDE PUPIL-TEACHER RATIO  
0                             NaN  
1                             NaN  
2                             NaN

** Condensing demographics **

In order to get unique values for `DBN`, we can get rid of other rows except for school year `20112012`. This will match up with our SAT results data.

In [28]:
data["demographics"] = data["demographics"][data["demographics"]["schoolyear"] == 20112012]
data["demographics"].head(3)

DBN                                              Name  schoolyear  \
6   01M015  P.S. 015 ROBERTO CLEMENTE                           20112012   
13  01M019  P.S. 019 ASHER LEVY                                 20112012   
20  01M020  PS 020 ANNA SILVER                                  20112012   

   fl_percent  frl_percent  total_enrollment prek    k grade1 grade2  \
6         NaN         89.4               189   13   31     35     28   
13        NaN         61.5               328   32   46     52     54   
20        NaN         92.5               626   52  102    121     87   

      ...     black_num black_per hispanic_num hispanic_per white_num  \
6     ...            63      33.3          109         57.7         4   
13    ...            81      24.7          158         48.2        28   
20    ...            55       8.8          357         57.0        16   

   white_per male_num male_per female_num female_per  
6        2.1     97.0     51.3       92.0       48.7  
13       8.5    147.0     44.8      181.0       55.2  
20       2.6    330.0     52.7      296.0       47.3  

[3 rows x 38 columns]

** Condensing Graduation **

Similar to demographics dataset, here we will pick the most recent available data for full cohort.

In [30]:
data["graduation"] = data["graduation"][data["graduation"]["Cohort"] == "2006"]
data["graduation"] = data["graduation"][data["graduation"]["Demographic"] == "Total Cohort"]

data["graduation"].head(3)

Demographic     DBN                            School Name Cohort  \
3   Total Cohort  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL   2006   
10  Total Cohort  01M448    UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   2006   
17  Total Cohort  01M450             EAST SIDE COMMUNITY SCHOOL   2006   

    Total Cohort Total Grads - n Total Grads - % of cohort Total Regents - n  \
3             78              43                     55.1%                36   
10           124              53                     42.7%                42   
17            90              70                     77.8%                67   

   Total Regents - % of cohort Total Regents - % of grads  \
3                        46.2%                      83.7%   
10                       33.9%                      79.2%   
17         74.400000000000006%                      95.7%   

              ...            Regents w/o Advanced - n  \
3             ...                                  36   
10            ...                                  34   
17            ...                                  67   

   Regents w/o Advanced - % of cohort Regents w/o Advanced - % of grads  \
3                               46.2%                             83.7%   
10                              27.4%                             64.2%   
17                74.400000000000006%                             95.7%   

   Local - n Local - % of cohort Local - % of grads Still Enrolled - n  \
3          7                  9%              16.3%                 16   
10        11                8.9%              20.8%                 46   
17         3                3.3%               4.3%                 15   

   Still Enrolled - % of cohort Dropped Out - n Dropped Out - % of cohort  
3                         20.5%              11                     14.1%  
10                        37.1%              20       16.100000000000001%  
17                        16.7%               5                      5.6%  

[3 rows x 23 columns]

Finally, let's convert AP scores data to `numeric` for easy comparison afterwards.

In [32]:
cols = ['AP Test Takers ', 'Total Exams Taken', 'Number of Exams with scores 3 4 or 5']

for col in cols:
    data["ap_2010"][col] = pd.to_numeric(data["ap_2010"][col], errors="coerce")
    
data["ap_2010"].head(3)

DBN                    SchoolName  AP Test Takers   Total Exams Taken  \
0  01M448  UNIVERSITY NEIGHBORHOOD H.S.             39.0               49.0   
1  01M450        EAST SIDE COMMUNITY HS             19.0               21.0   
2  01M515           LOWER EASTSIDE PREP             24.0               26.0   

   Number of Exams with scores 3 4 or 5  
0                                  10.0  
1                                   NaN  
2                                  24.0

** Merging all the datasets by performing joins **

AP scores and Graduation data may have missing DBN values, hence we perform `left` join with SAT results so as to retain `DBN` values.

In [37]:
combined = data["sat_results"]
combined = combined.merge(data["ap_2010"], on="DBN", how="left")
combined = combined.merge(data["graduation"], on="DBN", how="left")

combined.head(3)

DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   

  Num of SAT Test Takers  SAT Critical Reading Avg. Score  \
0                     29                            355.0   
1                     91                            383.0   
2                     70                            377.0   

   SAT Math Avg. Score  SAT Writing Avg. Score  sat_score  \
0                404.0                   363.0     1122.0   
1                423.0                   366.0     1172.0   
2                402.0                   370.0     1149.0   

                     SchoolName  AP Test Takers   Total Exams Taken  \
0                           NaN              NaN                NaN   
1  UNIVERSITY NEIGHBORHOOD H.S.             39.0               49.0   
2        EAST SIDE COMMUNITY HS             19.0               21.0   

             ...             Regents w/o Advanced - n  \
0            ...                                   36   
1            ...                                   34   
2            ...                                   67   

  Regents w/o Advanced - % of cohort Regents w/o Advanced - % of grads  \
0                              46.2%                             83.7%   
1                              27.4%                             64.2%   
2                74.400000000000006%                             95.7%   

  Local - n  Local - % of cohort Local - % of grads Still Enrolled - n  \
0         7                   9%              16.3%                 16   
1        11                 8.9%              20.8%                 46   
2         3                 3.3%               4.3%                 15   

  Still Enrolled - % of cohort Dropped Out - n Dropped Out - % of cohort  
0                        20.5%              11                     14.1%  
1                        37.1%              20       16.100000000000001%  
2                        16.7%               5                      5.6%  

[3 rows x 33 columns]

Since class size, demographics, survey and high school data have important information which we don't want to loose, we'll use `inner` join to combine them.

In [38]:
to_merge = ["class_size", "demographics", "survey", "hs_directory"]

for m in to_merge:
    combined = combined.merge(data[m], on="DBN", how="inner")

combined.head(3)

DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   

  Num of SAT Test Takers  SAT Critical Reading Avg. Score  \
0                     29                            355.0   
1                     91                            383.0   
2                     70                            377.0   

   SAT Math Avg. Score  SAT Writing Avg. Score  sat_score  \
0                404.0                   363.0     1122.0   
1                423.0                   366.0     1172.0   
2                402.0                   370.0     1149.0   

                     SchoolName  AP Test Takers   Total Exams Taken  \
0                           NaN              NaN                NaN   
1  UNIVERSITY NEIGHBORHOOD H.S.             39.0               49.0   
2        EAST SIDE COMMUNITY HS             19.0               21.0   

     ...                                   priority04  \
0    ...      Then to Manhattan students or residents   
1    ...                                          NaN   
2    ...                                          NaN   

                        priority05 priority06 priority07  priority08  \
0  Then to New York City residents        NaN        NaN         NaN   
1                              NaN        NaN        NaN         NaN   
2                              NaN        NaN        NaN         NaN   

  priority09 priority10                                         Location 1  \
0        NaN        NaN  220 Henry Street\nNew York, NY 10002\n(40.7137...   
1        NaN        NaN  200 Monroe Street\nNew York, NY 10002\n(40.712...   
2        NaN        NaN  420 East 12 Street\nNew York, NY 10009\n(40.72...   

         lat        lon  
0  40.713764 -73.985260  
1  40.712332 -73.984797  
2  40.729783 -73.983041  

[3 rows x 159 columns]

Filling in missing data due to `left` join with averages.

In [39]:
combined = combined.fillna(combined.mean())
combined = combined.fillna(0)

combined.head(3)

DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   

  Num of SAT Test Takers  SAT Critical Reading Avg. Score  \
0                     29                            355.0   
1                     91                            383.0   
2                     70                            377.0   

   SAT Math Avg. Score  SAT Writing Avg. Score  sat_score  \
0                404.0                   363.0     1122.0   
1                423.0                   366.0     1172.0   
2                402.0                   370.0     1149.0   

                     SchoolName  AP Test Takers   Total Exams Taken  \
0                             0       129.028846         197.038462   
1  UNIVERSITY NEIGHBORHOOD H.S.        39.000000          49.000000   
2        EAST SIDE COMMUNITY HS        19.000000          21.000000   

     ...                                   priority04  \
0    ...      Then to Manhattan students or residents   
1    ...                                            0   
2    ...                                            0   

                        priority05 priority06 priority07  priority08  \
0  Then to New York City residents          0          0           0   
1                                0          0          0           0   
2                                0          0          0           0   

  priority09 priority10                                         Location 1  \
0          0          0  220 Henry Street\nNew York, NY 10002\n(40.7137...   
1          0          0  200 Monroe Street\nNew York, NY 10002\n(40.712...   
2          0          0  420 East 12 Street\nNew York, NY 10009\n(40.72...   

         lat        lon  
0  40.713764 -73.985260  
1  40.712332 -73.984797  
2  40.729783 -73.983041  

[3 rows x 159 columns]

In [40]:
# Creating a district column to further help with our survey
def get_first_2_chars(name):
    return name[:2]
    
combined["school_dist"] = combined["DBN"].apply(get_first_2_chars)
combined["school_dist"].head(3)

0    01
1    01
2    01
Name: school_dist, dtype: object